### PR TITLE
Add obsoletion notification evilified-state-evilify

### PR DESCRIPTION
**Pull request by dalanicolai**
This PR simply adds the obsoletion notification as proposed in #14388.

### CODE REVIEW

**Comment by lebensterben** [comment]
There are a few invocations of this macro in our codebase.

See https://github.com/syl20bnr/spacemacs/search?q=%22evilified-state-evilify+%22&type=

**Comment by dalanicolai** [comment]
Generally, that does not matter for obsoletion, the notification is only to inform users to use the new macro in new code.
However, replacing the old macro invocations is not so much work, so I will quickly do that. I guess the old macro invocation could be removed completely then (if users use them in private packages, it is only 10 secs work to replace them).

File: layers/+readers/pdf/packages.el [hunk 1]
```diff
@@ -135,9 +154,9 @@
       (spacemacs/set-leader-keys-for-major-mode 'pdf-occur-buffer-mode
         "tf" 'next-error-follow-minor-mode))))
 
-(defun pdf/init-pdf-view-restore ()
-  (use-package pdf-view-restore
-    :after pdf-tools
-    :defer t
-    :init
-    (add-hook 'pdf-view-mode-hook 'pdf-view-restore-mode)))
+;; (defun pdf/init-pdf-view-restore ()
```
Comment (lebensterben) [inline layers/+readers/pdf/packages.el:157]: why is this commented out?

Comment (dalanicolai) [reply layers/+readers/pdf/packages.el:157]: Ah... that is an accident. I use this locally, while I am working on implementing/improving continuous scroll in pdf-tools...

File: layers/+emacs/helpful/packages.el [hunk 2]
```diff
@@ -36,9 +36,9 @@
       (add-hook 'emacs-startup-hook
                 (lambda ()
                   (spacemacs/set-leader-keys
-                    "hdk" #'helpful-key
-                    "hdf" #'helpful-callable
-                    "hdv" #'helpful-variable)
+                    "hhk" #'helpful-key
```
[the remark this answers is not in the document]
Comment (dalanicolai) [reply layers/+emacs/helpful/packages.el:39]: Ah, I thought I had skipped my personal settings. I prefer helpful not to overwrite the standard functionality (but I use this also locally, although I wanted to suggest that as a better keybinding in the helpful layer anyway).

Anyway, I think that while staged selectively, I somehow accidentally staged all when extending the commit (pressing `ce`). I will fix it. As far as I know, I only had personal changes in the helpful and pdf layers...

**Review by lebensterben — APPROVED** [review]

**Comment by dalanicolai** [comment]
Ah, I guess you did that already... thanks!

**Comment by lebensterben** [comment]
@dalanicolai 

>  Additionally, I have removed some redundant progns. AFAIK they are not required within the use-package form (you can check from expanding the use-package macro twice).

They are redundant. But we require this for unknown reason...
See https://develop.spacemacs.org/doc/CONVENTIONS.html#use-package

**Comment by dalanicolai** [comment]
I have again pushed another update. I encountered some more 'unknown variable: ...keymap` warnings (in the vc layer), so I have wrapped the relevant `evilified-state-evilify-map` forms within appropriate `with-eval-after-load` forms

**Comment by smile13241324** [comment]
@dalanicolai I have noticed there are some merge conflicts, care to have look?

**Comment by dalanicolai** [comment]
@smile Sure! Took a look, and there was conflict from some newer commit. Anyway, should be 'clean' now...

**Pull request merged**